src/journal/journal-file.h

```diff
@@ -85,6 +85,7 @@ typedef struct JournalFile {
         bool compress_lz4:1;
         bool seal:1;
         bool defrag_on_close:1;
+        volatile bool archive:1;
```

Comment:
Well, I'm inclined to agree that the volatile is unnecessary here, but there is a scenario where we may write to f->archive during an async offline which occurs in a separate thread.  But when that occurs, the f->offline_state will be operated on using atomics so it's probably safe to simply drop volatile and not introduce any other locks.

@poettering do you have any insights here?  I threw in the volatile to be defensive without putting much thought into it, since it looked a bit hairy, but it's important for this to be correct as it affects which state lands in the header during offline.